libass/ass.c

```diff
@@ -1524,12 +1524,27 @@ int ass_track_set_feature(ASS_Track *track, ASS_Feature feature, int enable)
 {
     switch (feature) {
     case ASS_FEATURE_INCOMPATIBLE_EXTENSIONS:
-        //-fallthrough
-#ifdef USE_FRIBIDI_EX_API
-    case ASS_FEATURE_BIDI_BRACKETS:
-        track->parser_priv->bidi_brackets = !!enable;
-#endif
+        #ifdef USE_FRIBIDI_EX_API
+            track->parser_priv->bidi_brackets = !!enable;
+        #endif
+        #ifdef CONFIG_UNIBREAK
+            track->parser_priv->wrap_unicode = !!enable;
+        #endif
         return 0;
+    case ASS_FEATURE_BIDI_BRACKETS:
+        #ifdef USE_FRIBIDI_EX_API
+            track->parser_priv->bidi_brackets = !!enable;
+            return 0;
+        #else
+            return -1;
+        #endif
```

Comment:
I agree that this should generally be left to #481, but I want to point out that this simpler pattern would be enough:

```c
#ifdef USE_FRIBIDI_EX_API
    case ASS_FEATURE_BIDI_BRACKETS:
        track->parser_priv->bidi_brackets = !!enable;
        return 0;
#endif
```

The default case handles the returning of −1, which is precisely the intention.

Reply:
Changed it to this simpler version.